File: benchmarkDS/CoT/CLAUDE_claude-sonnet-4-5-20250929/Lse120.c

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void allocateAndCopy(const char* data) {
    if (data == NULL) {
        printf("Error: Input data is NULL\\n");
        return;
    }
    
    // Limit size to prevent excessive memory allocation
    const size_t MAX_SIZE = 1024;
    size_t dataLen = strlen(data);
    
    if (dataLen > MAX_SIZE) {
        printf("Error: Input data exceeds maximum size\\n");
        return;
    }
    
    // Allocate first chunk with bounds checking
    char* chunk1 = (char*)malloc(dataLen + 1);
    if (chunk1 == NULL) {
        printf("Error: Memory allocation failed for chunk1\\n");
        return;
    }
    
    // Allocate second chunk with same size
    char* chunk2 = (char*)malloc(dataLen + 1);
    if (chunk2 == NULL) {
        printf("Error: Memory allocation failed for chunk2\\n");
        free(chunk1);
        return;
    }
    
    // Safely copy contents using secure function
    strncpy(chunk1, data, dataLen);
    chunk1[dataLen] = '\\0';
    
    strncpy(chunk2, chunk1, dataLen);
    chunk2[dataLen] = '\\0';
    
    // Print contents of second chunk
    printf("Contents of second chunk: %s\\n", chunk2);
    
    // Clean up memory
    free(chunk1);
    free(chunk2);
}

int main() {
    // Test case 1: Normal string
    printf("Test 1:\\n");
    allocateAndCopy("Hello World");
    
    // Test case 2: Empty string
    printf("\\nTest 2:\\n");
    allocateAndCopy("");
    
    // Test case 3: String with special characters
    printf("\\nTest 3:\\n");
    allocateAndCopy("Test@123!#$");
    
    // Test case 4: NULL input
    printf("\\nTest 4:\\n");
    allocateAndCopy(NULL);
    
    // Test case 5: Longer string
    printf("\\nTest 5:\\n");
    allocateAndCopy("Secure memory allocation test");
    
    return 0;
}

File: benchmarkDS/Vanilla/CLAUDE_claude-sonnet-4-5-20250929/Lse120.py

def copy_and_print(size, data):
    # Allocate two chunks of memory
    chunk1 = bytearray(size)
    chunk2 = bytearray(size)
    
    # Copy data into first chunk
    data_bytes = data.encode('utf-8')
    for i in range(min(len(data_bytes), size)):
        chunk1[i] = data_bytes[i]
    
    # Copy contents of first chunk into second chunk
    for i in range(size):
        chunk2[i] = chunk1[i]
    
    # Print contents of second chunk
    print(f"Contents of second chunk: {chunk2.decode('utf-8', errors='ignore')}")

if __name__ == "__main__":
    # Test case 1
    print("Test Case 1:")
    copy_and_print(10, "Hello")
    
    # Test case 2
    print("\\nTest Case 2:")
    copy_and_print(15, "World")
    
    # Test case 3
    print("\\nTest Case 3:")
    copy_and_print(20, "Java Programming")
    
    # Test case 4
    print("\\nTest Case 4:")
    copy_and_print(5, "Test")
    
    # Test case 5
    print("\\nTest Case 5:")
    copy_and_print(30, "Memory Allocation Example")
